### PR TITLE
Create generators modification after computation

### PR DESCRIPTION
**Pull request by FranckLecuyer**

### CODE REVIEW

**Review by jonenst — APPROVED** [review]

File: src/main/java/org/gridsuite/voltageinit/server/service/NetworkModificationService.java [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+        try {
+            VoltageInitModificationInfos voltageInitModificationInfos = new VoltageInitModificationInfos();
+
+            result.getGeneratorModifications().forEach(gm -> {
+                if (gm.getModifs().getTargetV() != null || gm.getModifs().getTargetQ() != null) {
+                    GeneratorModificationInfos.GeneratorModificationInfosBuilder builder = GeneratorModificationInfos.builder()
+                        .generatorId(gm.getGeneratorId());
+                    if (gm.getModifs().getTargetV() != null) {
+                        builder.voltageSetpoint(gm.getModifs().getTargetV());
```
Comment (sBouzols) [inline src/main/java/org/gridsuite/voltageinit/server/service/NetworkModificationService.java:86]: Could we use `@JsonInclude(Include.NON_NULL)` instead maybe ?

File: src/main/java/org/gridsuite/voltageinit/server/service/VoltageInitService.java [hunk 2]
```diff
@@ -67,10 +75,21 @@ private static VoltageInitResult fromEntity(VoltageInitResultEntity resultEntity
     }
 
     public void deleteResult(UUID resultUuid) {
+        Optional<VoltageInitResultEntity> result = resultRepository.find(resultUuid);
+        result.ifPresent(r -> {
+            if (r.getModificationsGroupUuid() != null) {
+                networkModificationService.deleteModificationsGroup(r.getModificationsGroupUuid());
```
Comment (sBouzols) [inline src/main/java/org/gridsuite/voltageinit/server/service/VoltageInitService.java:81]: Deo we have to wait this deletion before delete the result ?

File: src/main/java/org/gridsuite/voltageinit/server/VoltageInitController.java [hunk 3]
```diff
@@ -112,4 +112,14 @@ public ResponseEntity<Void> stop(@Parameter(description = "Result UUID") @PathVa
         voltageInitService.stop(resultUuid, receiver);
         return ResponseEntity.ok().build();
     }
+
+    @GetMapping(value = "/results/{resultUuid}/modifications-group", produces = APPLICATION_JSON_VALUE)
```
Comment (sBouzols) [inline src/main/java/org/gridsuite/voltageinit/server/VoltageInitController.java:116]: here we get the modifications group uuid not a resource, maybe we should do something for the path ?

**Review by sBouzols — COMMENTED** [review]
some comments

File: src/main/java/org/gridsuite/voltageinit/server/service/NetworkModificationService.java (hunk 1 above)
Comment (FranckLecuyer) [reply src/main/java/org/gridsuite/voltageinit/server/service/NetworkModificationService.java:86]: Done

File: src/main/java/org/gridsuite/voltageinit/server/service/VoltageInitService.java (hunk 2 above)
Comment (FranckLecuyer) [reply src/main/java/org/gridsuite/voltageinit/server/service/VoltageInitService.java:81]: No. I replaced it with an asynchronous call

File: src/main/java/org/gridsuite/voltageinit/server/VoltageInitController.java (hunk 3 above)
Comment (FranckLecuyer) [reply src/main/java/org/gridsuite/voltageinit/server/VoltageInitController.java:116]: Changed into /results/{resultUuid}/modifications-group-uuid

**Pull request merged**